kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/parquet/ArrayColumnReader.java

```diff
@@ -79,14 +89,16 @@ private static Converter createElementConverter(
 
         checkArgument(typeFromFile.getFieldCount() == 1,
                 "Expected exactly one field in the array type, but got: " + typeFromFile);
-        GroupType repeatedGroup = typeFromFile.getType(0).asGroupType();
-
-        // TODO: handle the legacy 2-level list physical format
-        checkArgument(repeatedGroup.getFieldCount() == 1,
+        Type type = typeFromFile.getType(0);
```

Comment:
Is this supporting a legacy list formats? If yes, I think there are more cases to handle. Also checkout the `ParquetFielReaderSuite.scala` (`read parquet file generated by parquet-thrift`) for disabled tests which can be enabled once this is fixed.

Also this can be a separate PR.

Reply:
Yes, here I just make a patch to work around the issue. We can open a new issue to trace the problem, I will attach the test table, it's easy to reproduce the issue.